packages/yarnpkg-nm/sources/hoist.ts

```diff
@@ -668,6 +684,9 @@ const hoistGraph = (tree: HoisterWorkTree, rootNodePath: Array<HoisterWorkTree>,
       }
     }
 
+    if (parentNode.dependencyKind === HoisterDependencyKind.EXTERNAL_SOFT_LINK && wereNodesHoisted)
```

Comment:
Why only EXTERNAL_SOFT_LINK need anotherRound? I meet a bug that parentNode is a sub workspace and could not hoist its dependencies.

Reply:
Please open an issue with a reproduction.

Reply:
#5332 here is the issue with a reproduction repo.